File: src/algorithms/balanced-string-split.js
/**
 * Given a balanced string s, returns the maximum number of balanced string
 * that can be obtained.
 * 
 * Balanced strings are those that have an equal quantity of 'L' and 'R'
 * characters.
 * 
 * @param {string} s - A string of uppercase 'L' and 'R' characters
 * @returns {number} - Returns a count of balanced strings
 */
function balancedStringSplit (s) {
  let res = 0;
  let leftCount = 0;
  let rightCount = 0;
  for (const char of s) {
    if (char === 'L') {
      leftCount++;
    } else {
      rightCount++;
    }
    if (leftCount === rightCount) {
      res++;
    }
  }
  return res;
}

File: src/algorithms/balanced-string-split.py
def balanced_string_split(s):
    """Balanced strings are those that have an equal quantity of 'L' and 'R' 
    characters. Given a balanced string s, split it into some number of 
    substrings such that each substring is balanced. Return the maximum number 
    of balanced strings you can obtain.

    Args:
        s (str): A string of uppercase "R" and "L" characters.

    Returns:
        int: Returns a count of balanced substrings.
    """
    count = 0
    l = 0
    r = 0
    for char in s:
        if char == "L":
            l += 1
        elif char == "R":
            r += 1
        if l == r:
            count += 1
    return count
